File: src/minecraft/net/machinemuse/utils/MuseMathUtils.scala
package net.machinemuse.utils

import java.util.Random

object MuseMathUtils {
  val random = new Random;

  def nextDouble: Double = random.nextDouble

  def clampDouble(value: Double, min: Double, max: Double): Double = {
    if (value < min) return min
    if (value > max) return max
    return value
  }
  def sumsq(x: Double, y: Double, z: Double) = x * x + y * y + z * z

  def pythag(x: Double, y: Double, z: Double) = Math.sqrt(x * x + y * y + z * z)
}

File: main/java/net/machinemuse/numina/math/MuseMathUtils.java
package net.machinemuse.numina.math;

import java.util.Random;

/**
 * Ported to Java by lehjr on 10/22/16.
 */
public final class MuseMathUtils {
    private static Random random = new Random();

    public static Random random() {
        return random;
    }

    public static double nextDouble() {
        return random().nextDouble();
    }

    public static double clampDouble(double value, double min, double max) {
        return value < min ? min : (value > max ? max : value);
    }

    public static double sumsq(double x, double y, double z) {
        return x * x + y * y + z * z;
    }

    public static double pythag(double x, double y, double z) {
        return Math.sqrt(x * x + y * y + z * z);
    }
}